lib/FakeDynamo.js

```diff
@@ -360,7 +362,7 @@ FakeTable.prototype.updateItem = function(data) {
 
   // done (ALL_OLD only returns ConsumedCapacity)
   return Q.resolve({
-    Attributes: oldItem,
+    Attributes: itemExists ? oldItem : null,
```

Comment:
This actually fixes the consistency issue.  Previously we were returning `oldItem` which had been updated with values that shouldn't exist on lines 349-352.  The additional check in the unit test is being run against `FakeDynamo` to double check that this behavior is correct.